core/workspace_comment.js

```diff
@@ -0,0 +1,370 @@
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.setWidth = function(width) {
+  this.width_ = width;
+};
+
+/**
+ * Get stored location.
+ * @return {!goog.math.Coordinate} The comment's stored location.  This is not
+ *     valid if the comment is currently being dragged.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.getXY = function() {
+  return this.xy_.clone();
+};
+
+/**
+ * Move a comment by a relative offset.
+ * @param {number} dx Horizontal offset, in workspace units.
+ * @param {number} dy Vertical offset, in workspace units.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.moveBy = function(dx, dy) {
+  var event = new Blockly.Events.CommentMove(this);
+  this.xy_.translate(dx, dy);
+  event.recordNew();
+  Blockly.Events.fire(event);
+};
+
+/**
+ * Get whether this comment is deletable or not.
+ * @return {boolean} True if deletable.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.isDeletable = function() {
+  return this.deletable_ &&
+      !(this.workspace && this.workspace.options.readOnly);
+};
+
+/**
+ * Set whether this comment is deletable or not.
+ * @param {boolean} deletable True if deletable.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.setDeletable = function(deletable) {
+  this.deletable_ = deletable;
+};
+
+/**
+ * Get whether this comment is movable or not.
+ * @return {boolean} True if movable.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.isMovable = function() {
+  return this.movable_ &&
+      !(this.workspace && this.workspace.options.readOnly);
+};
+
+/**
+ * Set whether this comment is movable or not.
+ * @param {boolean} movable True if movable.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.setMovable = function(movable) {
+  this.movable_ = movable;
+};
+
+/**
+ * Returns this comment's text.
+ * @return {string} Comment text.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.getContent = function() {
+  return this.content_;
+};
+
+/**
+ * Set this comment's content.
+ * @param {string} content Comment content.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.setContent = function(content) {
+  if (this.content_ != content) {
+    Blockly.Events.fire(
+        new Blockly.Events.CommentChange(this, this.content_, content));
+    this.content_ = content;
+  }
+};
+
+/**
+ * Encode a comment subtree as XML with XY coordinates.
+ * @param {boolean=} opt_noId True if the encoder should skip the comment id.
+ * @return {!Element} Tree of XML elements.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.toXmlWithXY = function(opt_noId) {
+  var element = this.toXml(opt_noId);
+  element.setAttribute('x', Math.round(this.xy_.x));
+  element.setAttribute('y', Math.round(this.xy_.y));
+  element.setAttribute('h', this.height_);
+  element.setAttribute('w', this.width_);
+  return element;
+};
+
+/**
+ * Encode a comment subtree as XML, but don't serialize the XY coordinates.
+ * This method avoids some expensive metrics-related calls that are made in
```

Comment:
Should the comments here also indicate that width and height are also not encoded here? And maybe another indication that they are encoded in the comments for `toXMLWithXY`